reports_automation/report_create.py

```diff
@@ -0,0 +1,118 @@
+import requests
+import os, json,sys
+from configparser import ConfigParser,ExtendedInterpolation
+
+sys.path.insert(0, '/opt/sparkjobs/ml-analytics-service/reports_automation/')
+from mongo_logging import *
+
+# Read the Config
+config_path = os.path.split(os.path.dirname(os.path.abspath(__file__)))
+config = ConfigParser(interpolation=ExtendedInterpolation())
+config.read(config_path[0] + "/report_config.ini")
+
+# Required field gathering for API
+base_url = config.get("API_ENDPOINTS","base_url")
+headers_api = {
+        'Content-Type': config.get("API_HEADERS", "content_type"),
+        'Authorization' : config.get("API_HEADERS","authorization_access_token")
+    }
+
+folder_path = config.get("REPORTS_FILEPATH","base_path") + config.get("REPORTS_FILEPATH","folder_name") + "/"
+
+# Creation of chart using Json config making an API call
+def backEnd_create(file_name):
+     try :
+        url_backend_create = base_url + config.get("API_ENDPOINTS","backend_create")
+        file_path = folder_path + "backEndConfig" + "/" + file_name
+        with open(file_path) as data_file:
+            json_config = json.load(data_file)
+            json_config["request"]["createdBy"] = config.get("JSON_VARIABLE","createdBy")
+            json_config["request"]["config"]["container"] = config.get("JSON_VARIABLE","container")
+            json_config["request"]["config"]["store"] = config.get("JSON_VARIABLE","store")
+            json_config["request"]["config"]["key"] = config.get("JSON_VARIABLE","key")
+            json_config["request"]["config"]["reportConfig"]["mergeConfig"]["postContainer"] = config.get("JSON_VARIABLE","container")
+            json_config["request"]["config"]["reportConfig"]["mergeConfig"]["container"] = config.get("JSON_VARIABLE","container")
+
+            doc = {
+                    "configFileName" : file_name,
+                    "config" : json.dumps(json_config),
+                    "operation": "backEnd_create"
+                  }
+
+        #Api call
+        response_api = requests.post(
```

Comment:
@Ashwiniev95 For a given file "**/migrations/releases/6.0.0/config/backend/update/1.json**" if it is success and the same config exists we should not execute that file in migration, we can skip that file

Reply:
Done